samples/TestApplication/Pages/Index.cshtml

```diff
@@ -98,35 +105,59 @@
                 Datasets = new List<ChartJsBarDataset>()
+            {
+                new ChartJsBarDataset()
                 {
-                    new ChartJsBarDataset()
-                    {
-                        Label = "# of Votes from blazor",
-                        BackgroundColor = new List<string>(){"#cc65fe" },
-                        BorderColor = "#cc65fe",
-                        PointHoverRadius = 2,
-                        Data = new List<double>(){ 19.187,12.2253,5.5,3,3,2}
-                    },
-                    new ChartJsBarDataset()
-                    {
-                        Label = "# of Likes from blazor",
-                        BackgroundColor = new List<string>() {
-                            "#a4cef0",
-                            "#3498db",
-                            "#95a5a6",
-                            "#9b59b6",
-                            "#f1c40f",
-                            "#e74c3c",
-                            "#34495e" },
-                        BorderColor = "#36a2eb",
-                        PointHoverRadius = 2,
-                        Data = new List<int>(){ 30,10,15,13,13,12}.Select<int,double>(i=> i).ToList()
-                    }
+                    Label = "# of Votes from blazor",
+                    BackgroundColor = new List<string>(){"#cc65fe" },
+                    BorderColor = "#cc65fe",
+                    PointHoverRadius = 2,
+                    Data = new List<double>(){ 19.187,12.2253,5.5,3,3,2}
+                },
+                new ChartJsBarDataset()
+                {
+                    Label = "# of Likes from blazor",
+                    BackgroundColor = new List<string>() {
+                        "#a4cef0",
+                        "#3498db",
+                        "#95a5a6",
+                        "#9b59b6",
+                        "#f1c40f",
+                        "#e74c3c",
+                        "#34495e" },
+                    BorderColor = "#36a2eb",
+                    PointHoverRadius = 2,
+                    Data = new List<int>(){ 30,10,15,13,13,12}.Select<int,double>(i=> i).ToList()
                 }
             }
+            },
+            //OnClick = (eventResult) =>
+            //{
+            //    if (eventResult.First().DatasetIndex != -1)
+            //    {
+            //        foreach (var item in eventResult)
+            //        {
+            //            _clickedValue += $"Dataset Index: {item.DatasetIndex}, Index: {item.Index} ";
+            //        }
+            //    }
+            //}
         };
     }
 
+    public async Task ProcessEvent(List<ChartJsEventResult> result)
+    {
+        clickedValue = "";
+        if (result.First().DatasetIndex != -1)
+        {
+            foreach (var item in result)
+            {
+                clickedValue += $"Dataset Index: {item.DatasetIndex}, Index: {item.Index} ";
+            }
+        }
+        StateHasChanged();
```

Comment:
`StateHasChanged()` is necessary at this point. Took an hour to figure it out. 😃